app/code/Magento/Store/Block/Switcher.php

```diff
@@ -234,4 +234,15 @@ public function getTargetStorePostData(Store $store, $data = [])
             $data
         );
     }
+
+    /**
+     * Returns target store url
+     *
+     * @param Store $store
+     * @return string
+     */
+    public function getStoreUrl(Store $store)
```

Comment:
hi @Mobecls 
Thank you for the pull requests.
Unfortunately this is prohibited code change that needs to be refactored https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/index.html#prohibited-code-changes

Reply:
Thank you for the answer! Have I broken the rule "Introduction of a method to a class or interface"? Should I create a new interface instead?